From 76c5e6b51768d8176335178fb4343700b52453e6 Mon Sep 17 00:00:00 2001
From: Mingming Yin <mingming@codeaurora.org>
Date: Thu, 17 Jul 2014 18:21:21 -0700
Subject: [PATCH] audiopolicy: Disable sonification on WFD sink

 * Aligned with CAF as of QSSI12, and removed unnecessary
   devices.isEmpty() check.

Original patch:
https://source.codeaurora.org/quic/la/platform/frameworks/av/commit/services/audiopolicy?h=LA.BR64.1.1.1-01010-8x16.0&id=9cc970fc512bedfa6a2cf03457d93609adc6eb85

Change-Id: I9bad6a294ddd7aee72f6f6a314666b892b730c8e
---
 services/audiopolicy/enginedefault/src/Engine.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/services/audiopolicy/enginedefault/src/Engine.cpp b/services/audiopolicy/enginedefault/src/Engine.cpp
index 08d64531c2..a54a93f577 100644
--- a/services/audiopolicy/enginedefault/src/Engine.cpp
+++ b/services/audiopolicy/enginedefault/src/Engine.cpp
@@ -394,6 +394,10 @@ DeviceVector Engine::getDevicesForStrategyInt(legacy_strategy strategy,
             devices2 = availableOutputDevices.getDevicesFromType(
                     AUDIO_DEVICE_OUT_ANLG_DOCK_HEADSET);
         }
+        if ((devices2.isEmpty()) && (strategy != STRATEGY_SONIFICATION)) {
+            // no sonification on WFD sink
+            devices2 = availableOutputDevices.getDevicesFromType(AUDIO_DEVICE_OUT_PROXY);
+        }
         if (devices2.isEmpty()) {
             devices2 = availableOutputDevices.getFirstDevicesFromTypes({
                         AUDIO_DEVICE_OUT_DGTL_DOCK_HEADSET, AUDIO_DEVICE_OUT_SPEAKER});
-- 
2.34.1

